include/flatbuffers/verifier.h

```diff
@@ -25,15 +25,27 @@ namespace flatbuffers {
 // Helper class to verify the integrity of a FlatBuffer
 class Verifier FLATBUFFERS_FINAL_CLASS {
  public:
+  struct Options {
+    uoffset_t max_depth = 64;
+    uoffset_t max_tables = 1000000;
+    bool check_alignment = true;
+    bool check_nested_flatbuffers = true;
+  };
+  Verifier(const uint8_t *const buf, const size_t buf_len, Options opts)
+      : Verifier(buf, buf_len, opts.max_depth, opts.max_tables,
+                 opts.check_alignment, opts.check_nested_flatbuffers) {}
+
   Verifier(const uint8_t *const buf, const size_t buf_len,
            const uoffset_t _max_depth = 64,
            const uoffset_t _max_tables = 1000000,
-           const bool _check_alignment = true)
+           const bool _check_alignment = true,
+           const bool _check_nested_flatbuffers = true)
```

Comment:
I would invert this, have this constructor call the new one you added.

Reply:
Oops, I just reread your PR description. I don't immediately see the challenge, but let me know more.

Reply:
Basically, my compiler was struggling to figure out what `Options { a, b, c d }` could possibly mean. Something about fields with default values made it stop creating default constructors. I just worked around it by manually implementing a the default constructor and a constructor given all the fields.